irpc.py

```diff
@@ -35,17 +36,31 @@ def hoist_declaration(main: FuncDef,
                     main.insert(0, node)
                     break
 
+def hoist_prov_memo(main: FuncDef,
+                    provdef: ProvDef):
+
+    _name_ = f"{e}_provided"
+    memo_node = Decl(name=_name_,
+        quals=[], storage=[],funcspec=[],
+        type=TypeDecl(declname=_name_, quals=[],
+                    type=IdentifierType(names=['bool'])),
+        init=ID(name='False'),
+        bitsize=None
+        )
```

Comment:
Can you create a variable for to store the `TypeDecl`? It will increase readability.

Reply:
I created variables for all of the parameters. Though it might make it easier to edit later on